code/modules/multiz/movement.dm

```diff
@@ -57,6 +58,20 @@
 			else
 				to_chat(src, "<span class='warning'>You gave up on pulling yourself up.</span>")
 				return 0
+		else if(catwalk?.hatch_open)
+			var/pull_up_time = max(5 SECONDS + (src.movement_delay() * 10), 1)
+			to_chat(src, "<span class='notice'>You grab the edge of \the [catwalk] and start pulling yourself upward...</span>")
+			var/old_dest = destination
+			destination = get_step(destination, dir) // mob's dir
+			if(!destination?.Enter(src, old_dest))
+				to_chat(src, "<span class='notice'>There's something in the way up above in that direction, try another.</span>")
+				return 0
+			destination.audible_message("<span class='notice'>You hear something climbing up \the [catwalk].</span>")
+			if(do_after(src, pull_up_time))
+				to_chat(src, "<span class='notice'>You pull yourself up.</span>")
```

Comment:
Is Enter() just a "can I do this" check, or will it actually trigger the `move()` sequence of calls? If it's just the former you appear to be missing a `forceMove()` or equivalent

Reply:
Enter is just a move check. It won't move objects to the turf. I am not sure if byond's Enter is that way, but like Move, we don't even call ..() in /turf/Enter so it doesn't matter since byond's behavior will never fire.

(Perhaps something to redo in the future. Paradise has Move() call ..() to do byond moves, but I don't know if it really matters much).